src/App.js

```diff
@@ -1,25 +1,71 @@
 import React from 'react';
-
 import './App.scss';
+import { Clock } from './clock';
+
+let interval;
+
+class App extends React.Component {
+  state = {
+    isClockVisible: false,
+    time: new Date().toLocaleTimeString(),
+    clockName: 0,
+  }
+
+  componentDidUpdate(_, prevState) {
+    clearInterval(interval);
```

Comment:
It should be in componentWillUnmount in Clock component. So every time clock is getting hidden - interval removed